### DISCUSSION TITLE
Getting error "⠋ Container snap-camera-server-300-mysql-1  Recreate                      0.0s  Error response from daemon: image with reference mysql:5.7 was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64"

### DISCUSSION THREAD

**Discussion by pokelonballs** [ptrumpis/snap-camera-server#23] [Q&A] [answered]
7:00 ish into the video here: https://www.youtube.com/watch?v=b2ILHJaD1T4

Getting this error.  Any ideas how i fix?

**Comment by ptrumpis** [comment]
Did you change your `docker-compose.yml` file?

https://github.com/ptrumpis/snap-camera-server/wiki/Docker-on-Apple-M1-(ARM64)

**Reply by pokelonballs** [reply]
yes i did that, but my system seems to be saying it is using amd64 not arm64.  I tried changing the same places to amd64 and that didn't work.  Any thoughts?

**Comment by ptrumpis** [comment]
Please share the content of your `docker-compose.yml`

**Reply by pokelonballs** [reply]
version: '3.8'

services:
  mysql:
    platform: linux/amd64
    image: mysql:5.7
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
    ports:
      - ${MYSQL_LOCAL_PORT}:${MYSQL_DOCKER_PORT}
    volumes:
      - db:/var/lib/mysql
      - ./mysql:/docker-entrypoint-initdb.d:ro

  adminer:
    platform: linux/amd64
    image: adminer
    environment:
      - ADMINER_DEFAULT_SERVER=mysql
      - ADMINER_DESIGN=${ADMINER_DESIGN}
    ports:
      - ${ADMINER_LOCAL_PORT}:${ADMINER_DOCKER_PORT}
    depends_on:
      - mysql

  webproxy:
    platform: linux/amd64
    image: nginx
    environment:
      - WEBAPP_PORT=${APP_DOCKER_PORT}
      - HTTP_PORT=${NGINX_DOCKER_PORT}
      - WWW_ROOT=${NGINX_WWW_ROOT}
    volumes:
      - ./ssl/studio-app.snapchat.com.crt:/etc/ssl/certs/studio-app.snapchat.com.crt:ro
      - ./ssl/studio-app.snapchat.com.key:/etc/ssl/private/studio-app.snapchat.com.key:ro
      - ./nginx/default.conf.template:/etc/nginx/templates/default.conf.template:ro
      - ./nginx/theme:/etc/nginx/theme:ro
      - ./nginx/snap-camera-categories-media:${NGINX_WWW_ROOT}/snap-camera-categories-media:ro
      - ./nginx/snap-camera-media:${NGINX_WWW_ROOT}/snap-camera-media:ro
      - ./nginx/snap-camera-media-alt:${NGINX_WWW_ROOT}/snap-camera-media-alt:ro
      - ./import:${NGINX_WWW_ROOT}/import
      - storage:${NGINX_WWW_ROOT}
    ports:
      - ${NGINX_LOCAL_PORT}:${NGINX_DOCKER_PORT}
      - 443:443
    depends_on:
      - webapp

  webapp:
    platform: linux/amd64
    build: .
    environment:
      - DB_HOST=mysql
      - DB_USER=${MYSQL_USER}
      - DB_PASS=${MYSQL_ROOT_PASSWORD}
      - DB_NAME=${MYSQL_DATABASE}
      - DB_PORT=${MYSQL_DOCKER_PORT}
      - PORT=${APP_DOCKER_PORT}
      - STORAGE_SERVER=http://studio-app.snapchat.com:${NGINX_LOCAL_PORT}
      - STORAGE_PATH=/var/lib/snapcamera
      - IMPORT_DIR=import
      - MEDIA_DIR=snap-camera-media
      - MEDIA_DIR_ALT=snap-camera-media-alt
    volumes:
      # content of snap-camera-media (MEDIA_DIR) is not getting included automatically through storage
      - ./nginx/snap-camera-media:/var/lib/snapcamera/snap-camera-media:ro
      - ./import:/var/lib/snapcamera/import
      - ./:/usr/src/app
      - /usr/src/app/node_modules
      - storage:/var/lib/snapcamera
    ports:
      - ${APP_LOCAL_PORT}:${APP_DOCKER_PORT}
    depends_on:
      - mysql

volumes:
  storage:
    name: server-storage
  db:
    name: mysql-db

**Comment by ptrumpis** [comment]
One of these solutions should work
- Try to run `docker-compose up --build`
- Try to remove the line `platform: linux/amd64` under `mysql:`

**Reply by pokelonballs** [reply]
appreciate your quick responses!  Here is the the response:

Error response from daemon: image with reference nginx was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8

**Reply by ptrumpis** [reply]
> Error response from daemon: image with reference nginx was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8

Remove the line `platform: linux/amd64` under `webproxy:`

**Reply by pokelonballs** [reply]
I removed that and also removed the line from mysql, tried it without removing that line from mysql:

Here was the response when the line was removed from both "webproxy" and "mysql" areas:

Error response from daemon: image with reference adminer was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8

Here is the response when the line is removed from just platform: "webproxy"

Error response from daemon: image with reference adminer was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8

**Reply by ptrumpis** [reply]
Ok lets try to go back to the beginning.

> Getting error "⠋ Container snap-camera-server-300-mysql-1 Recreate 0.0s Error response from daemon: image with reference mysql:5.7 was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64

Is that the original error message?
If so did that error occur before or after making changes to `docker-compose.yml`

The [Wiki Settings](https://github.com/ptrumpis/snap-camera-server/wiki/Docker-on-Apple-M1-(ARM64)) should always work on all Apple M1/M2 computers, but these settings may require running:

```sh
docker compose down
docker compose up --build
```

I don't have an explanation why the wiki settings shouldn't work, but in case they don't:

You only need the platform line under `mysql`, because it is referencing an old MySQL version without ARM support.
There is no need to have the platfom line anywhere else (but it shouldn't hurt either).

```yaml
services:
  mysql:
    platform: linux/amd64
    image: mysql:5.7
```

So I really don't understand the error message at the top.

**Here is what you can do:**

1. Try to manually delete all existing images in Docker Desktop.
2. Add the platform line only under `mysql:`
3. Open up a Terminal in the server directory and run:

```sh
docker compose down
docker compose up --build
```

If it still gives you an error please post that error message here.
The [Upgrade Guide](https://github.com/ptrumpis/snap-camera-server/wiki/Upgrading-to-Version-3.x#remove-the-image) contains a section about manually deleting stuff in Docker Desktop if you don't know how to do that.

**Comment by pokelonballs** [comment]
ok, doing this worked.  I also had to add the directory to access from Docker (which I hadn't done before), as well as delete all the images and the original container (I ran the compose up command multiple times, so that may have been confusing things).

Now I come across another challenge (i'm very untechnical btw so this is all stretching my capabilities (appreciate your patience).  I am following the video and am in the sudo part. I'm pressing ^O. When in the video you press ^O it says "wrote 11 lines" mine doesn't say anything.  It opens this section.  When I click the tab key on the mac (to complete...which I'm assuming is what we're going for) nothing happens either, so I'm thinking I"m doing something wrong (id10t error right? haha)?

I feel like we're getting soooo close.

<img width="582" alt="Screenshot 2023-04-20 at 9 14 16 PM" src="https://user-images.githubusercontent.com/90160819/233532479-2b0f7f80-70a1-43dc-8a13-5b3634363b19.png">

**Reply by ptrumpis** [reply, marked as answer]
It's asking you what file name to write. In other words where to save the file.
You need to hit Enter.

The actual key sequence is:
1. `^O`
2. `Enter`
3. `^X`

You can also download an /etc/hosts editor from the App Store with graphical user interface.

**Reply by pokelonballs** [reply]
I'm up and running!  A probably dumb question, but I'm assuming i have to keep docker open and running anytime i'm using snap camera right?  It's basically a localized server where the assets are stored and being pulled from?  Thank you again for this!

**Reply by ptrumpis** [reply]
correct
